apps/sms/views/shared/js/activity_handler.js

```diff
@@ -438,13 +441,19 @@ var ActivityHandler = {
                 text = slideArray[i].text;
                 break;
               }
-              text = text ? text : _('mms-message');
-              callback(text);
+
+              if (text) {
+                defer.resolve(text);
+              } else {
+                formatValue('mms-message').then(defer.resolve);
+              }
             });
           }
+
+          return defer.promise;
         }
 
-        Contacts.findByAddress(message.sender).then(function(contacts) {
+        return Contacts.findByAddress(message.sender).then(function(contacts) {
```

Comment:
Ah, sorry  that my comment was not clear. I mean if we use mozSetMessageHandlerPromise, then the unit test here will not need the change current `MockNavigatormozApps.mTriggerLastRequestSuccess` and we can start our test after give promise finished. Otherwise I'm not sure why we real need this API here.